src/modules/land_detector/land_detector_main.cpp

```diff
@@ -81,28 +81,30 @@ int LandDetector::task_spawn(int argc, char *argv[])
 
 	} else {
 		print_usage("unknown mode");
-		return -1;
+		return PX4_ERROR;
 	}
 
 	if (obj == nullptr) {
 		PX4_ERR("alloc failed");
-		return -1;
+		return PX4_ERROR;
 	}
 
+	// Remember current active mode
+	strncpy(_currentMode, argv[1], sizeof(_currentMode) - 1);
+	_currentMode[sizeof(_currentMode) - 1] = '\0';
+
+	_object.store(obj);
+	_task_id = task_id_is_work_queue;
 	int ret = obj->start();
 
 	if (ret < 0) {
+		_object.store(nullptr);
+		_task_id = -1;
 		delete obj;
```

Comment:
I know it's not a change in this PR, but, do we need to add an `obj = nullptr` here?